lib/services/autoupdates.go

```diff
@@ -96,4 +99,7 @@ type AutoUpdateService interface {
 
 	// DeleteAllAutoUpdateAgentReports deletes all AutoUpdateAgentReport resources.
 	DeleteAllAutoUpdateAgentReports(ctx context.Context) error
+
+	// SetAutoUpdateBotInstanceReport overwrites the singleton auto-update bot report.
+	SetAutoUpdateBotInstanceReport(ctx context.Context, spec *autoupdate.AutoUpdateBotInstanceReportSpec) error
```

Comment:
Even for singleton resources we want to use Get/Create/Update/Delete verbs [as per RFD 153](https://github.com/gravitational/teleport/blob/master/rfd/0153-resource-guidelines.md#api). This makes sure we stay consistent and always have a way to do conditional updates and delete the singleton.

Reply:
Gotcha. For now, I've renamed this method to `UpsertAutoUpdateBotInstanceReport` (because we want the unconditional overwrite semantics) and swapped the `spec` parameter for the full report object.

I've left the getter as-in (i.e. only accepting the context, not the name) to match `AuthPreference`, but if you think it's better to accept the name I'd be happy to do so in a later PR.